src/corefx/System.Globalization.Native/localeNumberData.cpp

```diff
@@ -0,0 +1,521 @@
+
+Implementation of NumberFormatInfo.PercentNegativePattern.
+Returns the pattern index.
+*/
+int GetPercentNegativePattern(const Locale &locale)
+{
+	const int DEFAULT_VALUE = 0;
+	static const char* Patterns[] = { "-n %", "-n%", "-%n", "%-n", "%n-", "n-%", "n%-", "-% n", "n %-", "% n-", "% -n", "n- %" };
+	UErrorCode status = U_ZERO_ERROR;
+
+	LocalPointer<NumberFormat> format(NumberFormat::createInstance(locale, UNUM_PERCENT, status));
+	assert(U_SUCCESS(status));
+	if (U_SUCCESS(status))
+	{
+		DecimalFormat* decimalFormat = dynamic_cast<DecimalFormat*>(format.getAlias());
+		assert(decimalFormat != NULL);
+		if (decimalFormat != NULL)
+		{
+			int value = GetNumericPattern(decimalFormat, Patterns, ARRAY_LENGTH(Patterns), true);
+			if (value >= 0)
+			{
+				return value;
+			}
+		}
+	}
+
+	return DEFAULT_VALUE;
+}
+
+/*
+Function:
+GetPercentPositivePattern
+
+Implementation of NumberFormatInfo.PercentPositivePattern.
+Returns the pattern index.
+*/
+int GetPercentPositivePattern(const Locale &locale)
+{
+	const int DEFAULT_VALUE = 0;
+	static const char* Patterns[] = { "n %", "n%", "%n", "% n" };
+	UErrorCode status = U_ZERO_ERROR;
+
+	LocalPointer<NumberFormat> format(NumberFormat::createInstance(locale, UNUM_PERCENT, status));
+	assert(U_SUCCESS(status));
+	if (U_SUCCESS(status))
+	{
+		DecimalFormat* decimalFormat = dynamic_cast<DecimalFormat*>(format.getAlias());
+		assert(decimalFormat != NULL);
+		if (decimalFormat != NULL)
+		{
+			int value = GetNumericPattern(decimalFormat, Patterns, ARRAY_LENGTH(Patterns), false);
+			if (value >= 0)
+			{
+				return value;
+			}
+		}
+	}
+
+	return DEFAULT_VALUE;
+}
+
+/*
+Function:
+GetMeasurementSystem
+
+Obtains the measurement system for the local, determining if US or metric.
+Returns 1 for US, 0 otherwise.
+*/
+UErrorCode GetMeasurementSystem(const char *localeId, int32_t *value)
+{
+	UErrorCode status = U_ZERO_ERROR;
+
+	UMeasurementSystem measurementSystem = ulocdata_getMeasurementSystem(localeId, &status);
+	if (U_SUCCESS(status))
+	{
+		*value = (measurementSystem == UMeasurementSystem::UMS_US) ? 1 : 0;
+	}
+
+	return status;
+}
+
+/*
+PAL Function:
+GetLocaleInfoInt
+
+Obtains integer locale information
+Returns 1 for success, 0 otherwise
+*/
+extern "C" int32_t GetLocaleInfoInt(const UChar* localeName, LocaleNumberData localeNumberData, int32_t* value)
+{
+	Locale locale = GetLocale(localeName);
+	if (locale.isBogus())
+	{
+		return UErrorCodeToBool(U_ILLEGAL_ARGUMENT_ERROR);
+	}
+
+	UErrorCode status = U_ZERO_ERROR;
+
+	switch (localeNumberData)
+	{
+	case LanguageId:
+		*value = locale.getLCID();
+		break;
+	case MeasurementSystem:
+		status = GetMeasurementSystem(locale.getName(), value);
+		break;
+	case FractionalDigitsCount:
+	{
+		UNumberFormat* numformat = unum_open(UNUM_DECIMAL, NULL, 0, locale.getName(), NULL, &status);
+		if (U_SUCCESS(status))
+		{
+			*value = unum_getAttribute(numformat, UNUM_MAX_FRACTION_DIGITS);
+			unum_close(numformat);
+		}
+		break;
+	}
```

Comment:
Going to defer this.  My understanding is that a style guide is coming for C++ code soon and I'm sure there will be other changes I need to make at the same time.